src/graph/connect.cc

```diff
@@ -626,7 +626,9 @@ ncclResult_t ncclTopoPostset(struct ncclComm* comm, int* firstRanks, int* treePa
 
   // Only use full MAXCHANNELS for gfx94x
   int maxChannels = IsArchMatch(comm->topo->nodes[GPU].nodes[0].gpu.gcn, "gfx94") ?
-    ((comm->topo->nodes[GPU].nodes[0].gpu.cu == 80 || comm->topo->nodes[GPU].nodes[0].gpu.cu == 20 || comm->topo->nodes[GPU].nodes[0].gpu.cu == 38)
+    ((comm->topo->nodes[GPU].nodes[0].gpu.cu == 38 ||
+      comm->topo->nodes[GPU].nodes[0].gpu.cu == 80 || comm->topo->nodes[GPU].nodes[0].gpu.cu == 20 ||
+      comm->topo->nodes[GPU].nodes[0].gpu.cu == 96 || comm->topo->nodes[GPU].nodes[0].gpu.cu == 24)
```

Comment:
With https://github.com/ROCm/rccl/pull/1310, we can use std::min(comm->topo->nodes[GPU].nodes[0].gpu.cu, MAXCHANNELS)